JavaBytecodeCompiler/src/main/java/org/openzen/zenscript/javabytecode/compiler/JavaExpressionVisitor.java

```diff
@@ -302,11 +302,12 @@ public Void visitCompare(CompareExpression expression) {
 					throw new UnsupportedOperationException("Unknown builtin comparator: " + expression.operator.getBuiltin());
 			}
 		} else {
-			if (!checkAndExecuteMethodInfo(expression.operator, expression.type, expression))
-				throw new IllegalStateException("Call target has no method info!");
-
 			expression.left.accept(this);
 			expression.right.accept(this);
+
+			if (!checkAndExecuteMethodInfo(expression.operator, expression.type, expression))
+				throw new IllegalStateException("Call target has no method info!");
+			
```

Comment:
I guess there's a reason for this, but why?

Reply:
Why the exception, or why this has been moved down?
The lines have been moved down because we usually first need the operands before we can call the method 😛